stdlib/public/Platform/libc-openbsd.modulemap

```diff
@@ -0,0 +1,26 @@
+//===--- libc-openbsd.modulemap ----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Partial modulemap for libc on OpenBSD.
+module SwiftGlibc [system] {
```

Comment:
At this point, why even have a separate OpenBSD modulemap, why not generate it from `glibc.modulemap.gyb` as the other platforms are doing, including my move to do the same with Bionic in #35707? I know you said "adding yet more conditionals to that modulemap impacts readability," but I don't see that simple file becoming a problem if you add OpenBSD to it.

Reply:
Sure, it's not too bad with only a few platforms supported, but as that grows, the conditionals become more complicated and the modulemap harder to reason about. Additionally, if we want to get to a point eventually where each libc has its own module, then we would probably want to have separate platform modulemaps anyway as it'll look cleaner in the wash.

I'm happy to do whatever, ultimately, and we can cross those bridges when we come to them.

Reply:
Personally, I'd prefer if you just used that common glibc gyb file for now, but not a big deal either way.

Reply:
@compnerd, wdyt about that option.

Reply:
What does the diff look like for having the extra conditionals in the shared gyb file? It doesn't really look like the platform has substantially different content to offer. Even in the future (cough LLVM), I would not expect a libc to diverge substantially from the layouts in that map.